### ISSUE TITLE
InMemoryKVCache has no observability or metrics for evictions

### ISSUE THREAD

**Issue by sachncs** [sachncs/avqa#32]
## Problem

`src/avqa/cache.py:125-130` evicts oldest tokens when `cache_size > max_size`:

```python
if self.max_size > 0 and self.size > self.max_size:
    excess = self.size - self.max_size
    self.cache_key = self.cache_key[..., excess:, :].contiguous()
    self.cache_value = self.cache_value[..., excess:, :].contiguous()
```

There is no logging, no metric, no callback for eviction events. Production users cannot tell whether their `max_size` is being hit, cannot A/B test different sizes, and cannot correlate downstream attention degradation with cache eviction.

## Evidence

- `src/avqa/cache.py:125-130` (quoted above).
- `src/avqa/profiling.py:190-196` `record_cache_hit` / `record_cache_miss` exist on the `Profiler` class but `InMemoryKVCache` never calls them.
- `src/avqa/cache.py` has no `logger` import (silent on eviction).

## Root Cause

The cache was written without observability hooks; the `Profiler` was added later (CHANGELOG 0.1.0 [Added]) but the integration point was missed.

## Impact

- Diagnosing production cache pressure requires patching the source.
- The existing `Profiler.record_cache_hit/miss` are dead code.
- The eviction is silent in production logs.

## Recommendation

1. Add a `logger = get_logger("cache")` import in `src/avqa/cache.py`.
2. Emit a `logger.debug("evicted %d tokens from KV cache (max_size=%d)", excess, self.max_size)` on eviction.
3. Add an `eviction_count` attribute (or a callback hook) that increments per eviction; expose it via `state_dict` or `cache_stats()`.
4. Wire `record_cache_hit/miss` calls from `InMemoryKVCache.lookup` and `append` (only when the optional `Profiler` is passed).

## Acceptance Criteria

- [ ] Eviction emits a debug log line.
- [ ] `cache_stats()` returns `{size, max_size, eviction_count, hit_count, miss_count}`.
- [ ] `tests/unit/test_cache.py` asserts the new attributes.

## Verification

Run a small loop that exceeds `max_size` and confirm the log line + eviction count.

## Risk

Low. Additive observability; no behavior change.

**Comment by sachncs** [comment]
Resolved in PR #37.
